src/doc/trpl/advanced-linking.md

```diff
@@ -0,0 +1,175 @@
+% Advanced Linking
+
+The common cases of linking with Rust have been covered earlier in this book,
+but supporting the range of linking possibilities made available by other
+languages is important for Rust to achieve seamless interaction with native
+libraries.
+
+# Native libraries required by upstream crates
+
+Sometimes an upstream Rust crate won't include an appropriate `#[link(...)]`
+attribute to automatically link against a required natve library on your system.
+This can happen if libraries are named differently across platforms (so only
+you as the crate consumer know the actual name). To link against the appropriate
+library for your system, you can use the `#[link(...)]` attribute on an empty
+extern block:
+
+``` no_run
+extern crate foo;
+
+#[link(name = "foonative3"]
+extern {}
+# fn main() {}
+```
+
+# Link args
+
+There is one other way to tell rustc how to customize linking, and that is via
+the `link_args` attribute. This attribute is applied to `extern` blocks and
+specifies raw flags which need to get passed to the linker when producing an
+artifact. An example usage would be:
+
+``` no_run
+#![feature(link_args)]
+
+#[link_args = "-foo -bar -baz"]
+extern {}
+# fn main() {}
+```
+
+Note that this feature is currently hidden behind the `feature(link_args)` gate
+because this is not a sanctioned way of performing linking. Right now rustc
+shells out to the system linker, so it makes sense to provide extra command line
+arguments, but this will not always be the case. In the future rustc may use
+LLVM directly to link native libraries in which case `link_args` will have no
+meaning.
```

Comment:
It may be worth also mentioning that there is a `-C link-args` argument, as well as the default linker being `gcc` on most systems but `link.exe` is used for MSVC.